ciao-4.11/contrib/lib/python3.5/site-packages/sherpa_contrib/chart.py

```diff
@@ -228,50 +234,74 @@ def save_chart_spectrum(outfile, clobber=True, verbose=True,
     """Plots the spectrum as used by ChART2.
 
-    See `get_chart_spectrum` for information on the
-    arguments supported by this function.
+    Parameters
+    ----------
+    id : int or string or None
+        If id is None then the default Sherpa id is used. This
+        dataset must have a grid and source model defined.
+    elow, ehigh, ewidth : number or None
+        Only used if all three are given, otherwise the grid
+        of the data set is used. The elo value is the start
+        of the low-energy bin, ehigh is the end of the high-energy
+        bin, and ewidth the bin width, all in keV.
+    norm : number or None
+        Multiply the flux values by this value, if set.
+    overplot : bool, optional
+        If ``True`` then the data is added to the current plot,
+        otherwise a new plot is created.
+    clearwindow: bool, optional
+        If ``True`` then clear out the current plot area of all
+        existing plots. This is not used if ``overplot`` is set.
 
     Notes
     -----
     A histogram plot is now used, instead of the curve plot
     used for ChART version 1.
+
+    Labels used to be added to the top-left and -right corners to
+    indicate the model and dataset identifier. The plot title
+    now contains the model name and the dataset identifier has been
+    removed.
     """
 
-    vals = _get_chart_spectrum(**kwargs)
-    c.open_undo_buffer()
-    try:
-        if c.info() is None:
-            c.add_window(["id", "chart"])
-        else:
-            c.erase()
-
-        c.add_histogram(vals["xlo"], vals["xhi"], vals["y"])
-        c.set_plot_xlabel("Energy (keV)")
-        c.set_plot_ylabel("Flux (photon cm^{-2} s^{-1})")
-        c.set_plot_title("ChaRT Source Spectrum")
-
-        id = c.ChipsId()
-        id.coord_sys = c.FRAME_NORM
-        c.add_label(id, 0.05, 0.97,
-                    "Model: {}".format(vals["model"]))
-        c.add_label(id, 0.95, 0.97,
-                    "Dataset: {}".format(repr(vals["id"])),
-                    ["halign", 1])
-
-    except:
-        c.discard_undo_buffer()
-        raise
-
-    c.close_undo_buffer()
+    vals = _get_chart_spectrum(id=id, elow=elow,
+                               ehigh=ehigh, ewidth=ewidth,
+                               norm=norm)
+
+    # In sherpa_contrib.utils.plot_instmap_weights it was found useful
+    # to convert the Y values to float32 from float64 to avoid
+    # precision issues confusing the plot (i.e. showing structure that
+    # isn't meaningful to the user). Is this going to be needed
+    # here too?
+    #
+    hplot = HistogramPlot()
+    hplot.xlo = vals['xlo']
+    hplot.xhi = vals['xhi']
+    hplot.y = vals['y']  # .astype(np.float32)
+    hplot.xlabel = 'Energy (keV)'
+    hplot.title = 'ChaRT Spectrum: {}'.format(vals['model'])
+
+    # LaTeX support depends on the backend
+    if backend.name == 'pylab':
+        hplot.ylabel = 'Flux (photon cm$^{-2}$ s$^{-1}$)'
+    elif backend.name == 'chips':
```

Comment:
Are we going to release this while chips is still alive?

Reply:
I think we should, and as it is a small code change to support I think it's worth it even if we don't (plus if I ever do decide to right a plotly / altair / ... backend for notebook use it reminds me to update this).